operations/helm/charts/grafana-agent/templates/controllers/_pod.yaml

```diff
@@ -57,6 +57,10 @@ spec:
   tolerations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+  {{- with .Values.controller.topologySpreadConstraints }}
+  topologySpreadConstraints:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```

Comment:
Should we only include this field if the value is set? (eg. similar to what we do with other fields like priorityClassName a few lines above)

Otherwise it looks like pods are failing to start as they're waiting for the empty conditions to be met.

Reply:
I looked at the tests after re-generating, and it looks like that whole section gets omitted unless the array is set to a non-empty value. In any case, I've set the condition to `ScheduleAnyway` in case there's some test I didn't know was being run during CI. That should get around any scheduling failure.